File: solutions/1496. Path Crossing/Solution.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        paths = [[0,0]]
        point = [0,0]
        for d in path:
            match d:
                case 'N':
                    point[1] += 1
                case 'S':
                    point[1] -= 1
                case 'E':
                    point[0] += 1
                case _:
                    point[0] -= 1
            if point in paths:
                return True
            else:
                paths.append(point[:])

File: solutions/1496. Path Crossing/Solution.ts
function isPathCrossing(path: string): boolean {
  let [x, y] = [0, 0];
  const paths: Set<number> = new Set();
  // Starting Point
  paths.add(0);
  for (const d of path) {
    if (d === "N") y++;
    else if (d === "S") y--;
    else if (d === "E") x++;
    else if (d === "W") x--;

    const point = x * 20000 + y; // 20000 Constraint OR any High No
    if (paths.has(point)) return true;
    paths.add(point);
  }
  return false;
}
